cocotb/bus.py

```diff
@@ -53,7 +53,7 @@ class Bus:
         Support for ``struct``/``record`` ports where signals are member names.
     """
 
-    def __init__(self, entity, name, signals, optional_signals=[], bus_separator="_", array_idx=None):
+    def __init__(self, entity, name, signals, optional_signals=[], bus_separator="_", case_insensitive=True, array_idx=None):
```

Comment:
```suggestion
    def __init__(self, entity, name, signals, optional_signals=[], bus_separator="_", case_insensitive=False, array_idx=None):
```

`False` is the current default, there is no need to be breaking interfaces over this. It could matter in case-sensitive languages.

Reply:
The logic does a case-sensitive check first, if that fails it falls back on case-insensitive.  So it will not break any existing tests even if case_insensitive=True by default.

Reply:
Currently if it fails case sensitive check, it would stop searching and warn instead of trying a case insensitive search (where it might pass without a warning in a corner case). That is a silent change in behavior. I have seen people lose an extraordinary amount of time to silent changes in behavior in corner cases, don't do it.

If this is the behavior you want I would make it tristate: explicit `True`, explicit `False`, and default. Default can be equivalent to `True` if that's what you think is best, but it must be explicit. I suggest a warning like "Found something with a case insensitive search that I otherwise wouldn't have. If you don't like that behavior, set to `False`."